### ISSUE TITLE
bug: Attribute onSelect is missing on InputHTMLAttributes and TextareaHTMLAttributes

### ISSUE THREAD

**Issue by cam-narzt** [ionic-team/stencil#4617]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.0.2

### Current Behavior

The onSelect attribute doesn't exist in the JSX types for the input and textarea elements, causing a failed build when using this attribute unless I use // @ts-expect-error to suppress the error.

### Expected Behavior

The attribute `onSelect` should exist for `JSXBase.InputHTMLAttributes<HTMLInputElement>` and `JSXBase.TextareaHTMLAttributes<HTMLTextAreaElement>` and be an event handling function with signature `(event: Event) => void`, causing there to be no TypeScript error when building the application.

### System Info

```shell
`npx stencil info` doesn't seem to work on node 20 or 18...


$ npx stencil info
/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935
 const r = e.process, n = createTerminalLogger({
             ^

TypeError: Cannot read properties of undefined (reading 'process')
    at exports.createNodeLogger (/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935:14)
    at Object.<anonymous> (/Users/camdennarzt/Developer/Javascript/stencil/bin/stencil:42:26)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47

Node.js v20.5.0
```

```
$ npx stencil info
/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935
 const r = e.process, n = createTerminalLogger({
             ^

TypeError: Cannot read properties of undefined (reading 'process')
    at exports.createNodeLogger (/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935:14)
    at Object.<anonymous> (/Users/camdennarzt/Developer/Javascript/stencil/bin/stencil:42:26)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47

Node.js v18.17.0
```
```


### Steps to Reproduce

Add `onSelect` attribute to an `<input>` or `<textarea>` element in a stencil component.

### Code Reproduction URL

https://github.com/cam-narzt/demo-stencil-bug

### Additional Information

I already created a PR to fix this: https://github.com/ionic-team/stencil/pull/4616

**Comment by rwaskiewicz** [comment]
@cam-narzt I had a few comments on the PR, and I think we can quickly resolve/discuss there.

Regarding `npx stencil info`, that's interesting 🤔 

I tried to reproduce the `npx stencil info` bug, but wasn't able to using node 20
```
➜  /tmp npm init stencil component info-test && cd $_
✔ All setup  in 543 ms

  We suggest that you begin by typing:

  $ cd info-test
  $ npm install
  $ npm start

  You may find the following commands will be helpful:

  $ npm start
    Starts the development server.

  $ npm run build
    Builds your project in production mode.

  $ npm test
    Starts the test runner.


  Further reading:

   - https://github.com/ionic-team/stencil-component-starter
   - https://stenciljs.com/docs

  Happy coding! 🎈

➜  info-test npm install
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.

added 409 packages, and audited 410 packages in 6s

36 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
➜  info-test npx stencil info

      System: node 20.5.0
    Platform: darwin (22.5.0)
   CPU Model: Apple M1 (8 cpus)
    Compiler: /tmp/info-test/node_modules/@stencil/core/compiler/stencil.js
       Build: 1690208707
     Stencil: 4.0.2 😈
  TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.19.1
``` 

Nor with 18:
```
➜  info-test volta install node@18
success: installed and set node@18.17.0 as default
   note: this version of Node includes npm@9.6.7, which is higher than your default version (8.19.4).
      To use the version included with Node, run `volta install npm@bundled`
➜  info-test npx stencil info

      System: node 18.17.0
    Platform: darwin (22.5.0)
   CPU Model: Apple M1 (8 cpus)
    Compiler: /private/tmp/info-test/node_modules/@stencil/core/compiler/stencil.js
       Build: 1690208707
     Stencil: 4.0.2 😈
  TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.19.1
```

Note - I'm using volta on my system, so it's possible that the shim is affecting my testing _somehow_.

Looking at the path in your stacktrace (`/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935`) is it possible that's a globally available version of Stencil? Does `npx stencil version` return '4.0.2' for you?

**Comment by cam-narzt** [comment]
```
$ npx stencil version
/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935
 const r = e.process, n = createTerminalLogger({
             ^

TypeError: Cannot read properties of undefined (reading 'process')
    at exports.createNodeLogger (/Users/camdennarzt/Developer/Javascript/stencil/sys/node/index.js:4935:14)
    at Object.<anonymous> (/Users/camdennarzt/Developer/Javascript/stencil/bin/stencil:42:26)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47

Node.js v20.5.0
```

**Comment by cam-narzt** [comment]
since the PR got merged i'll close this, i don't much care if `npx stencil info` works.

**Comment by tanner-reits** [comment]
This was resolved in [Stencil v4.0.3](https://github.com/ionic-team/stencil/releases/tag/v4.0.3). Please re-open this issue if you are still experiencing issues after updating.
